flips/20201125-bip-44-multi-account.md

```diff
@@ -5,192 +5,282 @@
+The legacy path is _deprecated_ by this FLIP, and should not be used to
+generate new keys.
 
-1. Derive the first account's node (index = 0).
-2. Scan for addresses by checking each public key index against the public key registry,
-respecting the [gap limit described in BIP 44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#address-gap-limit).
-3. If no address is found in the registry, stop discovery.
-4. If an address is found, query the chain to fetch the account details.
-5. If no account is found<sup>1</sup>, skip this index and go to step 1.
-6. If an account is found, add it to the user's wallet and go to step 1.
+The legacy path is:
+
+```
+m / 44' / 1' / 769 / 0 / 0
+```
+
+### Account Discovery
 
-<sup>1</sup>Flow supports account deletion, meaning that an address found in the registry
-may refer to a nonexistent account. In this case the account should be skipped
-but discovery should continue.
+Wallet developers should use a modified version of the original account
+discovery procedure [described in BIP
+44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#account-discovery).
+
+Modified constants from BIP 44 include:
+
+The key index gap limit should be `5`.
+The account index gap limit should be `5`.
+
+The prescribed account discovery procedure is as follows:
+
+1. Derive the key pair using the legacy path, checking it's use with the
+   public key registry. If an address is found, query the Flow network to fetch
+   the account's details. If an account is found, remember the relationship between the
+   path used to generate this key, and the account's details.
+2. Derive a key pair (starting with account index = 0 and key index =
+   1) using the path specified in this FLIP.
+3. Scan for the key usage by checking the public key registry.
+   If the gap limit has been reached for both key index and account index, stop
+   discovery.
+    3.1. If no address is found in the registry 
+        3.1.1 If the key index gap limit has been reached without finding any 
+        addressed in the registry, then go to step 2, incrementing the account index
+        by one and starting with key index = 0 again. 
+        3.1.2 If the key index gap limit has not been reached, go to step 2 and
+        increment the key index by one. 
+    3.2. If an address is found, query the Flow network to fetch the account's details. 
+        3.2.1. If no account is found<sup>1</sup>, go to step 2, incrementing 
+        the account index by one.
+        3.2.2. If an account is found, remember the relationship between the
+        path used to generate this key and the account's details. Then go to step 2, 
+        incrementing the key index by one.
+
+<sup>1</sup>Flow supports account deletion, meaning that an address found in the
+registry may refer to a nonexistent account. In this case the account should be
+skipped but discovery should continue.
```

Comment:
This is great! Should we also add a similar note for revoked keys? I mean we could precise if we consider a path leading to a revoked key, as a "used" path or a "free" path.

Reply:
I think it might depend on how the Public Key registry / Hardware Wallet API will have to work. If it returns back addresses for keys that are on the account but are also revoked, i'm less worried about confusion here than if it doesn't return addresses for such keys.

Reply:
Yep good point, I think this could be precised on the key registry side.